src/pudl/scripts/dbt_helper.py

```diff
@@ -166,6 +167,60 @@ def from_yaml(cls, schema_path: Path) -> "DbtSchema":
         with schema_path.open("r") as schema_yaml:
             return cls.model_validate(yaml.safe_load(schema_yaml))
 
+    @classmethod
+    def to_yaml(cls, schema_path: Path):
+        """Write DbtSchema object to YAML file."""
+        with schema_path.open("w") as schema_file:
+            yaml.dump(
+                cls.model_dump(exclude_none=True),
+                schema_file,
+                default_flow_style=False,
+                sort_keys=False,
+                width=float("inf"),
+            )
+
+
+def schema_has_removals_or_modifications(diff: DeepDiff) -> bool:
+    """Check if the DeepDiff includes any removals or modifications."""
+    change_keys = {
+        "values_changed",
+        "type_changes",
+        "dictionary_item_removed",
+        "iterable_item_removed",
+        "attribute_deleted",
+    }
+
+    return any(key in diff and diff[key] for key in change_keys)
+
+
+def _print_schema_diff(diff: DeepDiff, old_schema: DbtSchema, new_schema: DbtSchema):
+    """Print old and new YAML, and summary of schema changes."""
+    print("\n======================")
```

Comment:
non-blocking: Since this is intended only to happen interactively, using `print` should be fine here, though we could use the `logger` to print out these lines if we want to configure all the logging together...

also non-blocking: you could probably do this in one `print` call:

```python

print(
    "\n".join([
        "========",
        "Old YAML:",
        yaml.dump(...),
        ...
        _schema_diff_summary(diff) # returns a string now instead of printing out
    ])
)
```